artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ProtocolHandler.java

```diff
@@ -243,6 +243,18 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
          ctx.flush();
       }
 
+      @Override
+      public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+         // We definitely want to log the failure
+         ActiveMQServerLogger.LOGGER.failureDuringProtocolHandshake(ctx.channel().remoteAddress().toString(), cause);
+
+         // However, what else should be done here? I've laid out a few options...
+         ctx.fireChannelInactive();
+         ctx.close(); // this is done frequently in other implementations of exceptionCaught()
+         ctx.fireExceptionCaught(cause);
+         ctx.pipeline().remove(this); // this is done at the end of decode(); should we do this as well?
```

Comment:
If the outcome is terminal then you shouldn't need to remove this, depends on the handler but in general the pipeline is going to get torn down if you close the context.